.github/workflows/release-dry-run.yml

```diff
@@ -1,14 +1,39 @@
 name: Release (dry run)
 on:
-  pull_request_target:
+  pull_request:
     types:
       - opened
       - synchronize
 
+permissions:
+  contents: write
+
 jobs:
-  main:
-    name: '/'
-    uses: technology-studio/github-workflows/.github/workflows/_release.yml@main
-    with:
-      dry_run: true
-    secrets: inherit
+  release-dry-run:
+    name: 'Release (dry run)'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: technology-studio/github-workflows/.github/actions/github-app-token-generator@main
+        id: get-token
+        with:
+          app-id: ${{ secrets.TXO_GITHUB_BOT_APP_ID }}
+          private-key: ${{ secrets.TXO_GITHUB_BOT_APP_PRIVATE_KEY }}
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+      - uses: technology-studio/github-workflows/.github/actions/install-dependencies@main
+      - run: yarn test --coverage
+      - uses: codecov/codecov-action@b9fd7d16f6d7d1b5d2bec1a2887e65ceed900238 # v4
+      - run: |
+          echo "PR_HEAD_REF=${{ github.event.pull_request.head.ref }}" >> $GITHUB_ENV
+      - run: GITHUB_ACTIONS="" yarn semantic-release --extends ./.releaserc.dryrun.js
+        env:
+          GITHUB_TOKEN: ${{ steps.get-token.outputs.token }}
```

Comment:
do we need to use github app bot token, can we use the provisioned token instead ?